databases/metadata/src/main/functions/04_ontology_hierarchy.sql

```diff
@@ -0,0 +1,32 @@
+SET search_path = public, pg_catalog;
+
+--
+-- A function that returns all of the ontology subclasses of the class with the given
+-- ontology_version and IRI.
+--
+CREATE OR REPLACE FUNCTION ontology_class_hierarchy(VARCHAR, VARCHAR)
```

Comment:
Ah, I should have been more specific, though once I say it I think you'll see why: the details of plans hide the details inside of functions, just showing "Function Scan", which is why I like to see how the internals of the functions perform separately.

But, given your same values and knowing where there's a dev DB set up, I can do it myself!:

```
                                                                                              QUERY PLAN                                                                                              
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 CTE Scan on subclasses  (cost=2490.71..2500.53 rows=491 width=96) (actual time=0.781..87.847 rows=738 loops=1)
   CTE subclasses
     ->  Recursive Union  (cost=0.82..2490.71 rows=491 width=98) (actual time=0.775..85.447 rows=738 loops=1)
           ->  Limit  (cost=0.82..155.94 rows=1 width=98) (actual time=0.768..0.769 rows=1 loops=1)
                 ->  Nested Loop Left Join  (cost=0.82..155.94 rows=1 width=98) (actual time=0.764..0.764 rows=1 loops=1)
                       Join Filter: ((h.subclass_iri)::text = (c.iri)::text)
                       ->  Index Scan using ontology_classes_pkey on ontology_classes c  (cost=0.41..8.43 rows=1 width=61) (actual time=0.172..0.172 rows=1 loops=1)
                             Index Cond: (((ontology_version)::text = 'http://edamontology.org/1.14/2016/04/20/20-39-12.5'::text) AND ((iri)::text = 'http://edamontology.org/operation_0004'::text))
                       ->  Index Only Scan using ontology_hierarchies_pkey on ontology_hierarchies h  (cost=0.41..147.50 rows=1 width=74) (actual time=0.585..0.585 rows=0 loops=1)
                             Index Cond: ((ontology_version = 'http://edamontology.org/1.14/2016/04/20/20-39-12.5'::text) AND (subclass_iri = 'http://edamontology.org/operation_0004'::text))
                             Heap Fetches: 0
           ->  Nested Loop  (cost=0.82..232.49 rows=49 width=98) (actual time=0.186..10.119 rows=138 loops=8)
                 ->  Nested Loop  (cost=0.41..125.78 rows=45 width=74) (actual time=0.146..4.165 rows=138 loops=8)
                       ->  WorkTable Scan on subclasses sc  (cost=0.00..0.20 rows=10 width=32) (actual time=0.001..0.117 rows=92 loops=8)
                       ->  Index Only Scan using ontology_hierarchies_pkey on ontology_hierarchies h_1  (cost=0.41..12.52 rows=4 width=74) (actual time=0.035..0.037 rows=1 loops=738)
                             Index Cond: ((ontology_version = 'http://edamontology.org/1.14/2016/04/20/20-39-12.5'::text) AND (class_iri = (sc.iri)::text))
                             Heap Fetches: 0
                 ->  Index Scan using ontology_classes_pkey on ontology_classes c_1  (cost=0.41..2.36 rows=1 width=61) (actual time=0.036..0.038 rows=1 loops=1103)
                       Index Cond: (((ontology_version)::text = 'http://edamontology.org/1.14/2016/04/20/20-39-12.5'::text) AND ((iri)::text = (h_1.subclass_iri)::text))
 Planning time: 3.589 ms
 Execution time: 89.081 ms
(21 rows)
```

Also at https://explain.depesz.com/s/abMk;

I'll look at this and then add any extra comments in an extra comment in a bit.